lib/coverband/adapters/redis_store.rb

```diff
@@ -22,7 +22,13 @@ def initialize(redis, opts = {})
       end
 
       def clear!
```

Comment:
I think I introduced a thread safety issue here in that we only have one instance of our store per ruby process. If two different threads call `clear!` at the same time we could have unintended results since we are depending on the the `type` state of the store.

I had started by passing the type as a method argument but it required a lot of passing that type state from method to method and didn't feel right. 

We could solve this by either creating a separate store per thread or introducing a Mutex. I think introducing a separate store per thread might be overkill considering then we would probably want a separate redis connection per thread?

Reply:
yeah I want to refactor the way the store works with 2 types, good point on the thread issue. When I refactor it to only be a concept known about by the store I will try to address.

Reply:
I haven't resolved this yet

Reply:
resolved the race conditions